var/spack/repos/builtin/packages/acts/package.py

```diff
@@ -145,8 +145,8 @@ class Acts(CMakePackage, CudaPackage):
     depends_on('boost @1.62:1.69 +program_options +test', when='@:0.10.3')
     depends_on('boost @1.71: +filesystem +program_options +test', when='@0.10.4:')
     depends_on('cmake @3.14:', type='build')
-    depends_on('dd4hep @1.11:', when='+dd4hep')
-    depends_on('dd4hep @1.11: +geant4', when='+dd4hep +geant4')
+    depends_on('dd4hep @1.11: +dddetectors +ddrec', when='+dd4hep')
+    depends_on('dd4hep @1.11: +dddetectors +ddrec +ddg4', when='+dd4hep +geant4 +examples')
```

Comment:
If you have `+dd4hep+geant4~examples`, does this not require `dd4hep+geant4`? More precisely, since it looks like this corrects an issue where `dd4hep` doesn't have a `geant4` variant, would `acts+geant4~examples` not require `dd4hep+ddg4`? FWIW either way you can simplify the above to 

```
depends_on('dd4hep @1.11: +dddetectors +ddrec', when='+dd4hep')
depends_on('dd4hep +ddg4', when='+dd4hep +geant4 +examples')
```

(these constraints will combine)

Reply:
> If you have `+dd4hep+geant4~examples`, does this not require `dd4hep+geant4`?

It indeed doesn't require geant4 in that case, since the geant4 dependence is only needed for the examples (`ACTS_BUILD_EXAMPLES_GEANT4` in acts CMakeLists.txt).

Thanks for the other suggestion. That simplifies the logic.